tests/test_core.py

```diff
@@ -5629,14 +5629,20 @@ def test_dlmalloc_inline(self):
 
   @no_asan('depends on the specifics of memory size, which for asan we are forced to increase')
   def test_dlmalloc(self):
-    self.banned_js_engines = [config.NODE_JS] # slower, and fail on 64-bit
+#    self.banned_js_engines = [config.NODE_JS] # slower, and fail on 64-bit
     # needed with typed arrays
     self.set_setting('INITIAL_MEMORY', '128mb')
 
     # Linked version
     self.do_runf(path_from_root('tests', 'dlmalloc_test.c'), '*1,0*', args=['200', '1'])
     self.do_run('dlmalloc_test.js', '*400,0*', args=['400', '400'], no_build=True)
 
+    # Large allocation should fail
+    old_args = self.emcc_args[:]
+    self.emcc_args += ['-s', 'ABORTING_MALLOC=0']
+    self.do_runf(path_from_root('tests', 'dlmalloc_test_large.c'), '0 0 0 1')
+    self.emcc_args = old_args
```

Comment:
if adding a new `self.do_runf` then it might as well be in another python function. That avoids the lines with `old_args` and makes testing it easier.

however, I was hoping that perhaps we can save even that, and just add the 5 lines in the new test to an existing dlmalloc test? they have no side effects (if the test passes) so there seems no downside to appending to an existing test, and as a benefit it will be faster in the test suite.

Reply:
Malloc will crash in the test unless compiled with ABORTING_MALLOC=0, so cannot add to an existing test .c file unless I also change the flags that is built with. Also now I realize that this should actually be tested with ALLOW_MEMORY_GROWTH enabled as well. Separated to a new test.